include/golioth/ota.h

```diff
@@ -188,6 +188,25 @@ enum golioth_status golioth_ota_download_component(struct golioth_client *client
                                                    ota_component_block_write_cb cb,
                                                    void *arg);
 
+/// Verify the component sha256 hash supplied by server matches on calculated locally
+///
+/// This function compares the component hash received in the OTA manifest to one that is supplied.
+/// Use the golioth_sys_sha256_*() functions (or similar) to calculate the has as blocks are
+/// downloaded, and pass the result to the \p local_hash parameter of this function.
+///
+/// @param component The @ref golioth_ota_component pointer from the OTA manifest
+/// @param local_hash An array, exactly 32 bytes in length, containing a locally-calculated sha256
+/// hash for the component that was downloaded.
+///
+/// @retval GOLIOTH_OK The \p local_hash matches the component hash recevied from the server.
+/// server.
+/// @retval GOLIOTH_ERR_INVALID_FORMAT The \p local_hash does not match the hash the server
+/// supplied.
+/// @retval GOLIOTH_ERR_NULL One of the supplied parameters was NULL.
+enum golioth_status golioth_ota_component_verify_sha256(
```

Comment:
Why create this new API instead of just having `golioth_ota_download_component()` calculate and verify the SHA automatically?

Reply:
The user may want to verify the hash of the full component after the block download has completed and the data is written somewhere in memory. I would think this is for users who want to roll their own component management outside of the automated download (with resume) that we will eventually have in the OTA service.

Do you prefer it be removed?

Reply:
If the user wants to verify the hash, they can extract it from the manifest and do it themselves. This is a simple utility function, I don't think it makes sense to have it as part of our API

Reply:
Sounds good to me. I've removed this commit and will rename the PR.